File: getuserdata.py
import requests
import csv
import sys


def getToken(userId, deviceId):
	header = {
		'x-api-key': 'kjsdkajdhkajhsdaksjdhaksjdhakjsdhq98w7eyq98we',
		'Content-Type': 'application/json'
	}

	data = { 
		"userId": userId, 
		"deviceId": deviceId,
		"deviceInfo": { "kind": "web", "model": "", "sdkVersion": "2.0" },
		"displayName": userId
	}

	endpoint = 'https://qatest.amity.co/v1/device'
	
	r = requests.post(url = endpoint, data = data, headers = header)
	return r.json()['token']


filename = './data/userdata.csv'
outputfile = './data/userdata-token.csv'

with open(filename, mode='r') as inputcsv, open(outputfile, mode='w') as outputcsv:
    csv_reader = csv.DictReader(inputcsv)

    alldata = []
    for row in csv_reader:
    	print("Getting token for " + row['userId'] + " - " + row['deviceId'])
    	row['token'] = getToken(row['userId'], row['deviceId'])
    	alldata.append(row)

    print("Writing output to " + outputfile)
    writer = csv.DictWriter(outputcsv, fieldnames = csv_reader.fieldnames)
    writer.writeheader()
    writer.writerows(alldata)
    

File: getuserdata.js
const axios = require('axios');
const fs = require('fs');
const csv = require('csvtojson');
const { Parser } = require('json2csv');
const fields = ['userId', 'deviceId', 'token'];
const json2csvParser = new Parser({ fields });

let results = [];
let promises = [];
let filename = './data/userdata.csv';
let outputfile = './data/userdata-token.csv';
let url = 'https://qatest.amity.co/v1/device'

const option = {
  header: {
    'x-api-key': 'kjsdkajdhkajhsdaksjdhaksjdhakjsdhq98w7eyq98we',
    'Content-Type': 'application/json'    
  }
}

const csvconverter = new Promise((resolve,reject) => {
  csv()
  .fromFile(filename)
  .then((json)=>{
      resolve(json);
  })
});

csvconverter.then(results => {
  results.forEach(data => {

    const reqbody = { 
        userId: data['userId'], 
        deviceId: data['deviceId'],
        deviceInfo: { "kind": "web", "model": "", "sdkVersion": "2.0" },
        displayName: data['userId']
      }

    promises.push(
      axios.post(url, reqbody, option)
      .then((res) => {
          data['token'] = res.data['token'];
      }).catch((err) => {
          console.error(err);
      }));
    });

  Promise.all(promises).then(() => {
    const csv = json2csvParser.parse(results);

    fs.writeFile(outputfile, csv, function(err) {
      if (err) throw err;
      console.log("Token generated: " + outputfile);
    })


  });

});
